### ISSUE TITLE
linter: tsconfig `paths` are ignored when the config declaring them is extended from a subdirectory

### ISSUE THREAD

**Issue by lozinsky** [oxc-project/oxc#25790]
### What version of Oxlint are you using?

1.78.0

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
import { defineConfig } from 'oxlint';

export default defineConfig({
  plugins: ['import'],
  categories: {
    correctness: 'off',
  },
  rules: {
    'import/no-cycle': 'error',
  },
});
```

### What happened?

Reproduction: https://github.com/lozinsky/oxlint-tsconfig-extends-subdir
CI logs: https://github.com/lozinsky/oxlint-tsconfig-extends-subdir/actions/runs/32019670175/job/95356504360#step:5:7

Two source files import each other through the `@/*` alias, so `import/no-cycle` must report a cycle:

```ts
// src/a.ts
import '@/b';
```

```ts
// src/b.ts
import '@/a';
```

The alias is declared in a base config that is pulled in via `extends`. Each case below writes those config files, then runs `tsc --noEmit` and `oxlint` in the project root. Only the location of the base config, and which config declares `include`, differ.

**Case 1: base config in the project root:**

```jsonc
// tsconfig.json
{
  "extends": "./tsconfig.base.json",
}
```

```jsonc
// tsconfig.base.json
{
  "compilerOptions": {
    "paths": {
      "@/*": ["./src/*"],
    },
  },
  "include": ["./src"],
}
```

```
tsc:    OK (paths resolved)
oxlint: 2 cycle errors
```

**Case 2: the same base config moved to a subdirectory (its relative paths adjusted accordingly):**

```jsonc
// tsconfig.json
{
  "extends": "./subdir/tsconfig.base.json",
}
```

```jsonc
// subdir/tsconfig.base.json
{
  "compilerOptions": {
    "paths": {
      "@/*": ["../src/*"],
    },
  },
  "include": ["../src"],
}
```

```
tsc:    OK (paths resolved)
oxlint: no cycle errors   <-- unexpected
```

`tsc --showConfig` for this case resolves to `"include": ["subdir/../src"]` with `files` `./src/a.ts`, `./src/b.ts`, i.e. TypeScript treats it exactly like case 1. oxlint reports nothing at all - `@/b` and `@/a` behave as if no tsconfig were found.

**Case 3: case 2 with the tsconfig passed explicitly (`oxlint --tsconfig=./tsconfig.json`, the very same root config that auto-discovery would pick):**

```
tsc:    OK (paths resolved)
oxlint: 2 cycle errors
```

**Case 4 — case 2 with `include` declared in the root config instead of the base one:**

```jsonc
// tsconfig.json
{
  "extends": "./subdir/tsconfig.base.json",
  "include": ["./src"],
}
```

```jsonc
// subdir/tsconfig.base.json
{
  "compilerOptions": {
    "paths": {
      "@/*": ["../src/*"],
    },
  },
  "include": ["../src"],
}
```

```
tsc:    OK (paths resolved)
oxlint: 2 cycle errors
```

So `paths` inherited from a subdirectory are applied correctly as soon as either `include` comes from the root config (case 4) or the tsconfig is passed explicitly (case 3) - only the plain `extends` into a subdirectory (case 2) silently loses them.

Expected: oxlint reports the cycle in every case, like `tsc` resolves the paths in every case.
